site/tests/app/libraries/plagiarism/IntervalTester.php

```diff
@@ -7,18 +7,85 @@
 use app\libraries\plagiarism\Interval;
 
 class IntervalTester extends \PHPUnit\Framework\TestCase {
-    public function testInterval() {
-        $interval = new Interval(1, 2, "match");
-        $this->assertSame(1, $interval->getStart());
-        $this->assertSame(2, $interval->getEnd());
-        $this->assertEmpty($interval->getOthers());
-        $interval->updateStart(3);
-        $interval->updateEnd(5);
-        $this->assertSame(3, $interval->getStart());
-        $this->assertSame(5, $interval->getEnd());
-
-        $this->assertSame("match", $interval->getType());
-        $interval->updateType("specific-match");
-        $this->assertSame("specific-match", $interval->getType());
+
+    private $my_interval;
+
+    public function setUp(): void {
+        $this->my_interval = [
+            new Interval(13, 19, "match"),
+            new Interval(73, 84, "common"),
+            new Interval(34, 42, "provided"),
+        ];
+    }
+
+    public function testGetters() {
+        $this->assertEquals($this->my_interval[0]->getStart(), 13);
+        $this->assertEquals($this->my_interval[0]->getEnd(), 19);
+        $this->assertEquals($this->my_interval[0]->getType(), "match");
+
+        $this->assertEquals($this->my_interval[1]->getStart(), 73);
+        $this->assertEquals($this->my_interval[1]->getEnd(), 84);
+        $this->assertEquals($this->my_interval[1]->getType(), "common");
+
+        $this->assertEquals($this->my_interval[2]->getStart(), 34);
+        $this->assertEquals($this->my_interval[2]->getEnd(), 42);
+        $this->assertEquals($this->my_interval[2]->getType(), "provided");
+    }
+
+    public function testSetters() {
+        $this->my_interval[0]->updateStart(15);
+        $this->assertEquals($this->my_interval[0]->getStart(), 15);
+
+        $this->my_interval[0]->updateEnd(22);
+        $this->assertEquals($this->my_interval[0]->getEnd(), 22);
+
+        $this->my_interval[0]->updateType("specific-match");
+        $this->assertEquals($this->my_interval[0]->getType(), "specific-match");
+    }
+
+    public function testOtherMatchingPositions() {
+        // TODO: should the interval class still support input matching
+        // regions with no start and end positions?
```

Comment:
Yes, we still need to support this.  It's perhaps a little confusing the way we have it currently but this is how the matching users popup list is generated.  If you look in `Interval.php` on line 68, it's adding a user+version+source gradeable which is a match but doesn't have start and end positions because the user+version+source gradeable is not the current user on the right and thus doesn't need them.